android/app/src/main/res/layout/toolbar_search_controls.xml

```diff
@@ -20,20 +20,35 @@
     tools:src="@drawable/ic_down"
     android:contentDescription="@string/back"/>
 
-  <com.google.android.material.textfield.TextInputEditText
-    android:id="@+id/query"
-    style="@style/MwmTextAppearance.Toolbar.Search"
-    android:layout_width="0dp"
-    android:layout_height="wrap_content"
+  <com.google.android.material.textfield.TextInputLayout
+    android:id="@+id/query_container"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
     android:layout_weight="1"
-    android:background="@null"
-    android:hint="@string/search"
-    android:imeOptions="actionSearch|flagNoExtractUi"
-    android:inputType="text"
-    android:padding="@dimen/margin_half_plus"
-    android:singleLine="true"
-    android:textCursorDrawable="@drawable/cursor_drawable"
-    tools:ignore="UnusedAttribute"/>
+    android:layout_gravity="center_vertical"
+    app:endIconMode="clear_text"
+    app:endIconTint="@android:color/white"
+    style="@style/Widget.MaterialComponents.TextInputLayout.FilledBox.Dense"
+    app:boxBackgroundMode="none"
+    app:hintEnabled="false">
+
+    <com.google.android.material.textfield.TextInputEditText
+        android:id="@+id/query"
+        style="@style/MwmTextAppearance.Toolbar.Search"
```

Comment:
```suggestion
    style="@style/MwmTextAppearance.Toolbar.Search"
    app:boxBackgroundMode="none"
    app:hintEnabled="false">

    <com.google.android.material.textfield.TextInputEditText
        android:id="@+id/query"
```

Removed style from `TextInputEditText`, updated style in `TextInputLayout`

Your change
<img width="322" height="102" alt="Screenshot 2025-08-25 at 16 29 19" src="https://github.com/user-attachments/assets/8fda348b-cb76-4e99-b933-0432ce2fc351" />

Fixed
<img width="288" height="127" alt="fixed" src="https://github.com/user-attachments/assets/382471ea-3e7a-459b-9ba4-ec76456353f8" />